cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamCoordinator.java

```diff
@@ -54,4 +55,14 @@
    * @return A {@link Cancellable} to cancel the watch
    */
   Cancellable addListener(String streamName, StreamPropertyListener listener);
+
+  /**
+   * Affect a Stream handler leader to a stream.
+   *
+   * @param streamName name of the stream.
+   * @return A {@link ListenableFuture} describing the progress of the operation.
+   */
+  ListenableFuture<Void> affectLeader(String streamName);
```

Comment:
It's better name it as `streadCreated` to abstract out the implementation (on stream creation, there can be more than just changing leader).